sklearnex/spmd/covariance/tests/test_incremental_covariance_spmd.py

```diff
@@ -0,0 +1,177 @@
+# ==============================================================================
+# Copyright 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import numpy as np
+import pytest
+from numpy.testing import assert_allclose
+
+from onedal.tests.utils._dataframes_support import (
+    _convert_to_dataframe,
+    get_dataframes_and_queues,
+)
+from sklearnex.tests._utils_spmd import (
+    _generate_statistic_data,
+    _get_local_tensor,
+    _mpi_libs_and_gpu_available,
+)
+
+
+@pytest.mark.skipif(
+    not _mpi_libs_and_gpu_available,
+    reason="GPU device and MPI libs required for test",
+)
+@pytest.mark.parametrize(
+    "dataframe,queue",
+    get_dataframes_and_queues(dataframe_filter_="dpnp,dpctl", device_filter_="gpu"),
+)
+@pytest.mark.parametrize("assume_centered", [True, False])
+@pytest.mark.mpi
+def test_incremental_covariance_fit_spmd_gold(dataframe, queue, assume_centered):
+    # Import spmd and batch algo
+    from sklearnex.covariance import IncrementalEmpiricalCovariance
+    from sklearnex.spmd.covariance import (
+        IncrementalEmpiricalCovariance as IncrementalEmpiricalCovariance_SPMD,
+    )
+
+    # Create gold data and process into dpt
+    data = np.array(
+        [
+            [0.0, 0.0, 0.0],
+            [0.0, 1.0, 2.0],
+            [0.0, 2.0, 4.0],
+            [0.0, 3.0, 8.0],
+            [0.0, 4.0, 16.0],
+            [0.0, 5.0, 32.0],
+            [0.0, 6.0, 64.0],
+            [0.0, 7.0, 128.0],
+        ]
+    )
```

Comment:
https://github.com/intel/scikit-learn-intelex/pull/1961#discussion_r1732087952

If this is being used as a default 'gold_data' for all testing for more estimators you may want to move it into _utils_spmd.py

Reply:
https://github.com/intel/scikit-learn-intelex/pull/1961#discussion_r1732638526